google-map-poly.html

```diff
@@ -0,0 +1,554 @@
+        }
+      }
+    },
+
+    _closedChanged: function() {
+      this._mapChanged();
+    },
+
+    _dragEventsChanged: function() {
+      if (this.poly) {
+        if (this.clickEvents) {
+          this._forwardEvent('drag');
+          this._forwardEvent('dragend');
+          this._forwardEvent('dragstart');
+        } else {
+          this._clearListener('drag');
+          this._clearListener('dragend');
+          this._clearListener('dragstart');
+        }
+      }
+    },
+
+    _editableChanged: function() {
+      this.poly && this.poly.setEditable(this.editable);
+    },
+
+    _fillColorChanged: function() {
+      this.poly && this.poly.set('fillColor', this.fillColor);
+    },
+
+    _fillOpacityChanged: function() {
+      this.poly && this.poly.set('fillOpacity', this.fillOpacity);
+    },
+
+    _geodesicChanged: function() {
+      this.poly && this.poly.set('geodesic', this.geodesic);
+    },
+
+    _iconsChanged: function() {
+      this.poly && this.poly.set('icons', this.icons);
+    },
+
+    _mapChanged: function() {
+      // Poly will be rebuilt, so disconnect existing one from old map and listeners.
+      if (this.poly) {
+        this.poly.setMap(null);
+        google.maps.event.clearInstanceListeners(this.poly);
+      }
+
+      if (this.map && this.map instanceof google.maps.Map) {
+        this._createPoly();
+      }
+    },
+
+    _mouseEventsChanged: function() {
+      if (this.poly) {
+        if (this.mouseEvents) {
+          this._forwardEvent('mousedown');
+          this._forwardEvent('mousemove');
+          this._forwardEvent('mouseout');
+          this._forwardEvent('mouseover');
+          this._forwardEvent('mouseup');
+        } else {
+          this._clearListener('mousedown');
+          this._clearListener('mousemove');
+          this._clearListener('mouseout');
+          this._clearListener('mouseover');
+          this._clearListener('mouseup');
+        }
+      }
+    },
+
+    _strokeColorChanged: function() {
+      this.poly && this.poly.set('strokeColor', this.strokeColor);
+    },
+
+    _strokeOpacityChanged: function() {
+      this.poly && this.poly.set('strokeOpacity', this.strokeOpacity);
+    },
+
+    _strokePositionChanged: function() {
+      this.poly && this.poly.set('strokePosition', this._convertStrokePosition());
+    },
+
+    _strokeWeightChanged: function() {
+      this.poly && this.poly.set('strokeWeight', this.strokeWeight);
+    },
+
+    _zIndexChanged: function() {
+      this.poly && this.poly.set('zIndex', this.zIndex);
+    },
+
+    // Helper logic.
+
+    _buildPathFromPoints: function() {
+      this._points = Array.prototype.slice.call(Polymer.dom(this.$.points).getDistributedNodes());
+      window.console.log("Points:  " + this._points.length);
+      window.console.log(Polymer.dom(this).getDestinationInsertionPoints());
+
+      // Build path from current points (ignoring vertex insertions while doing so).
+      this._building = true;
```

Comment:
Do you really need this? This operation should be super fast and you're only using this before when someone tries to edit.